## Fibonacci sequence 
### Step-by-step solution

In [1]:
# 1. Define the list with first 2 elements.

fib = [0, 1]

In [2]:
# 2. Calculate the value of next element.

x = fib[0]
y = fib[1]
z = x + y

print(z)

1


In [3]:
# 3. Append the number to list.

fib.append(z)

In [4]:
print(fib)

[0, 1, 1]


In [7]:
# 4. Create a loop that iterate through the last and penultimate values, sum them and append the sum to the list up to 100.

for i in range(2, 100):
    z = fib[i - 1] + fib[i - 2]
    fib.append(z)

### Complete

In [9]:
fib = [0, 1]

for i in range(100):
    fib.append(fib[-1] + fib[-2])
print(fib)

[0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946, 17711, 28657, 46368, 75025, 121393, 196418, 317811, 514229, 832040, 1346269, 2178309, 3524578, 5702887, 9227465, 14930352, 24157817, 39088169, 63245986, 102334155, 165580141, 267914296, 433494437, 701408733, 1134903170, 1836311903, 2971215073, 4807526976, 7778742049, 12586269025, 20365011074, 32951280099, 53316291173, 86267571272, 139583862445, 225851433717, 365435296162, 591286729879, 956722026041, 1548008755920, 2504730781961, 4052739537881, 6557470319842, 10610209857723, 17167680177565, 27777890035288, 44945570212853, 72723460248141, 117669030460994, 190392490709135, 308061521170129, 498454011879264, 806515533049393, 1304969544928657, 2111485077978050, 3416454622906707, 5527939700884757, 8944394323791464, 14472334024676221, 23416728348467685, 37889062373143906, 61305790721611591, 99194853094755497, 160500643816367088, 259695496911122585, 420196140727489673, 679891637638612258, 110008777

In [10]:
len(fib)

102

The sequence is correct, but it has more elements that expected

In the first line of code, we already assigned 2 values to the list. This means that, if we loop 100 times, we will end up with 102 values.

We should stop looping at 98

In [11]:
fib = [0, 1]

for i in range(98):
    fib.append(fib[-1] + fib[-2])

print(fib)

[0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946, 17711, 28657, 46368, 75025, 121393, 196418, 317811, 514229, 832040, 1346269, 2178309, 3524578, 5702887, 9227465, 14930352, 24157817, 39088169, 63245986, 102334155, 165580141, 267914296, 433494437, 701408733, 1134903170, 1836311903, 2971215073, 4807526976, 7778742049, 12586269025, 20365011074, 32951280099, 53316291173, 86267571272, 139583862445, 225851433717, 365435296162, 591286729879, 956722026041, 1548008755920, 2504730781961, 4052739537881, 6557470319842, 10610209857723, 17167680177565, 27777890035288, 44945570212853, 72723460248141, 117669030460994, 190392490709135, 308061521170129, 498454011879264, 806515533049393, 1304969544928657, 2111485077978050, 3416454622906707, 5527939700884757, 8944394323791464, 14472334024676221, 23416728348467685, 37889062373143906, 61305790721611591, 99194853094755497, 160500643816367088, 259695496911122585, 420196140727489673, 679891637638612258, 110008777

In [12]:
len(fib)

100

#### Important to remember:

Machine is never wrong, it just follows your commands, which may be wrong.